### PR TITLE
[12.0][ADD] hr_holidays_validate_days: check for outdated number of days requested

### PR DESCRIPTION
**Pull request by alexey-pelykh**
This module allows leave's number of days validation to handle cases when
changes to Resource Calendar (e.g. Global Leaves or Timezone) may change
computed number of days.

To validate quantities of requested days:
#. Go to *Leaves > Managers > All > Leaves*.
#. Select leaves of interest and perform *Action > Validate Number of Days*.

### CODE REVIEW

File: hr_holidays_validate_days/models/hr_leave.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class HrLeave(models.Model):
+    _inherit = 'hr.leave'
+
+    @api.multi
+    def action_validate_number_of_days(self):
+        def validate_number_of_days(leave):
```
Comment (tarteo) [inline hr_holidays_validate_days/models/hr_leave.py]: Can you put this method or the domain into class level so it can be inherited.

**Review by tarteo — CHANGES_REQUESTED** [review]
I don't know if it's too much work but it would be nice if the result is put into a wizard. So now or later it can be extended with a button like 'Fix outdated leaves'

File: hr_holidays_validate_days/models/hr_leave.py (hunk 1 above)
Comment (alexey-pelykh) [reply hr_holidays_validate_days/models/hr_leave.py]: Sure, done

**Comment by alexey-pelykh** [comment]
@tarteo fixing is complicated - it requires unvalidate=>reset to draft=>....=>validate done by bot

**Review by tarteo — APPROVED** [review]
Thanks!

**Review by yweng8111 — APPROVED** [review]
Nice Job, Thanks for your contribution.

**Comment by alexey-pelykh** [comment]
@pedrobaeza do you have any comments on the subject?

**Comment by pedrobaeza** [comment]
By my part, we shouldn't add more plurals in the module names if avoidable. `hr_holidays` prefix is imposed by Odoo, but `days` can be replaced by the singular. Seeing the goal of the module, I would call it `hr_holidays_length_verification`, and more if you request hours, not days, in your leave :wink:

**Comment by alexey-pelykh** [comment]
@pedrobaeza agree on plurals, `validation` vs `verification` vs `check`?

**Comment by alexey-pelykh** [comment]
I'm leaning towards `validate_length`, only `length_check` is shorter

**Comment by pedrobaeza** [comment]
`length_validation` then. It's preferable to use nouns instead of verbs. Don't worry about the length of the module, that it's not too much yet.

**Comment by alexey-pelykh** [comment]
@pedrobaeza done

**Comment by pedrobaeza** [comment]
README doesn't reflect the question about the days/hours duality, but if you don't mind, me neither.

**Comment by alexey-pelykh** [comment]
@pedrobaeza you mentioned, I changed 👍

**Comment by pedrobaeza** [comment]
Hehe, very compliant!

**Pull request merged**